src/libs/Navigation/linkTo.js

```diff
@@ -5,6 +5,43 @@ import linkingConfig from './linkingConfig';
 import getTopmostReportId from './getTopmostReportId';
 import getStateFromPath from './getStateFromPath';
 
+/**
+ * Motivation for this function is described in NAVIGATION.md
+ *
+ * @param {Object} action action generated by getActionFromState
+ * @param {Object} state The root state
+ * @returns {{minimalAction: Object, targetName: String}} minimal action is the action that we should dispatch and targetName is the name of the target navigator.
+ * targetName name is necessary to determine if we are going to use REPLACE for navigating between RHP flows.
+ */
+function getMinimalAction(action, state) {
+    let currentAction = action;
+    let currentState = state;
+    let currentTargetKey = null;
+    let targetName = null;
+
+    while (currentState.routes[currentState.index].name === currentAction.payload.name) {
+        if (!currentState.routes[currentState.index].state) {
+            break;
+        }
+
+        targetName = currentState.routes[currentState.index].name;
+        currentState = currentState.routes[currentState.index].state;
+
+        currentTargetKey = currentState.key;
+
+        // Creating new smaller action
+        currentAction = {
+            type: currentAction.type,
+            payload: {
+                name: currentAction.payload.params.screen,
+                params: currentAction.payload.params.params,
```

Comment:
```suggestion
                params: currentAction.payload.params.params,
                path: currentAction.payload.params.path,
```
path was missing here and it caused regression - https://github.com/Expensify/App/issues/22308

Reply:
@situchan Can you raise a PR fixing this?